README.md
# NotifyMe
NotifyMe jest wtyczką do komunikatora [AQQ](http://www.aqq.eu/pl.php). Powiadamia o sprawdzaniu naszej wersji oprogramowania oraz ostatniej aktywności przez innego użytkownika.

### Wymagania
Do skompilowania wtyczki NotifyMe potrzebne jest:

* Embarcadero RAD Studio XE7 (C++Builder oraz Delphi, jeżeli chcemy kompilować wersję x64)
* Pełna wersja komponentów [AlphaControls](http://www.alphaskins.com/) (do kompilacji wersji x64 potrzebne jest Delphi)
* [Plik nagłówkowy SDK komunikatora AQQ](https://bitbucket.org/beherit/pluginapi-for-aqq-im)
* Opcjonalnie [UPX](http://upx.sourceforge.net/) dla zmniejszenia rozmiaru pliku wynikowego (w szczególności wersji x64)

### Błędy
Znalezione błędy wtyczki należy zgłaszać w [tym temacie](http://forum.aqq.eu/topic/12041-notifyme/) znajdującym na oficjalnym forum komunikatora AQQ lub pisząc bezpośrednio do autora wtyczki (preferowany kontakt poprzez Jabber).

### Kontakt z autorem
Autorem wtyczki NotifyMe jest Krzysztof Grochocki. Możesz skontaktować się z nim drogą mailową (kontakt@beherit.pl) lub poprzez Jabber (im@beherit.pl).

### Licencja
Wtyczka NotifyMe objęta jest licencją [GNU General Public License 3](http://www.gnu.org/copyleft/gpl.html).

~~~~
NotifyMe
Copyright (C) 2013-2015  Krzysztof Grochocki

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
~~~~

CHANGELOG.md
1.2.0.2
-----
* Drobna aktualizacja angielskiej lokalizacji.

1.2.0.0
-----
* Obsługa lokalizacji (polskiej i angielskiej).
* Pełne wsparcie kolorystyki stylizacji okien.
* Pokazywanie notyfikacji w oknie rozmowy.
* Poprawki zmiany kart przy aktywnym pokazywaniu szczegółów statystyk.
* Poprawki w filtrze antyspamowym.
* Aktualizacja komponentów AlphaControls do nowszej wersji (9.19).

1.1.6.0
-----
* Nieskórkowanie systemowych okien dialogowych niepochodzących z wtyczki.
* Aktualizacja komponentów AlphaControls do nowszej wersji (9.10).

1.1.5.0
-----
* Lepsza prezentacja nazwy kontaktu w przypadku jego braku na liście.
* Poprawki w enumeracji listy kontaktów.
* Aktualizacja komponentów AlphaControls do nowszej wersji (9.05).

1.1.4.0
-----
* Ignorowanie pakietów wysyłanych przez sam serwer.

1.1.3.0
-----
* Możliwość włączenia przycisku szybkiego dostępu do statystyk.
* Poprawki w przebudowie pliku XML ze statystykami.
* Inne małe poprawki na formie ustawień.

1.1.2.0
-----
* Poprawki w pobieraniu pseudonimów kontaktów.
* Małe zmiany na formie ustawień.
* Zmniejszenie rozmiaru wtyczki 64-bitowej poprzez użycie kompresji LZMA.
* Aktualizacja komponentów AlphaControls do nowszej wersji (8.53).

1.1.0.0
-----
* Dodanie zabezpieczenia przed spamowaniem chmurkami.
* Wersja 64-bitowa jest wreszcie prawidłowo skórkowana.
* Aktualizacja komponentów AlphaControls do nowszej wersji (8.46).

1.0.0.0
-----
* Pierwsza publiczna wersja.

NotifyMe.h
//---------------------------------------------------------------------------
// Copyright (C) 2013-2014 Krzysztof Grochocki
//
// This file is part of NotifyMe
//
// NotifyMe is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3, or (at your option)
// any later version.
//
// NotifyMe is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with GNU Radio; see the file COPYING. If not, write to
// the Free Software Foundation, Inc., 51 Franklin Street,
// Boston, MA 02110-1301, USA.
//---------------------------------------------------------------------------

#include <vcl.h>
#include <tchar.h>


SettingsFrm.h
//---------------------------------------------------------------------------
// Copyright (C) 2013-2014 Krzysztof Grochocki
//
// This file is part of NotifyMe
//
// NotifyMe is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3, or (at your option)
// any later version.
//
// NotifyMe is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with GNU Radio; see the file COPYING. If not, write to
// the Free Software Foundation, Inc., 51 Franklin Street,
// Boston, MA 02110-1301, USA.
//---------------------------------------------------------------------------

//---------------------------------------------------------------------------
#ifndef SettingsFrmH
#define SettingsFrmH
#define WM_ALPHAWINDOWS (WM_USER + 666)
//---------------------------------------------------------------------------
#include "acPNG.hpp"
#include "acProgressBar.hpp"
#include "sBevel.hpp"
#include "sButton.hpp"
#include "sCheckBox.hpp"
#include "sEdit.hpp"
#include "sLabel.hpp"
#include "sListView.hpp"
#include "sPageControl.hpp"
#include "sPanel.hpp"
#include "sSkinManager.hpp"
#include "sSkinProvider.hpp"
#include "sSpinEdit.hpp"
#include <IdBaseComponent.hpp>
#include <IdThreadComponent.hpp>
#include <System.Actions.hpp>
#include <System.Classes.hpp>
#include <Vcl.ActnList.hpp>
#include <Vcl.ComCtrls.hpp>
#include <Vcl.Controls.hpp>
#include <Vcl.ExtCtrls.hpp>
#include <Vcl.Menus.hpp>
#include <Vcl.StdCtrls.hpp>
//---------------------------------------------------------------------------
class TSettingsForm : public TForm
{
__published:	// IDE-managed Components
	TsSkinManager *sSkinManager;
	TsSkinProvider *sSkinProvider;
	TActionList *ActionList;
	TAction *aExit;
	TAction *aLoadSettings;
	TAction *aSaveSettings;
	TsBevel *Bevel;
	TsButton *SaveButton;
	TsButton *CancelButton;
	TsButton *OKButton;
	TsPageControl *sPageControl;
	TsTabSheet *SettingsTabSheet;
	TsTabSheet *StatsTabSheet;
	TsLabel *NotificationLabel;
	TsCheckBox *OnVersionCheckBox;
	TsCheckBox *OnLastCheckBox;
	TsSpinEdit *CloudTimeOutSpinEdit;
	TsCheckBox *StatsCheckBox;
	TAction *aAllowSave;
	TsListView *sListView;
	TsButton *ReloadButton;
	TsButton *DeleteButton;
	TAction *aGetDataFromXML;
	TsCheckBox *ExInfoCheckBox;
	TAction *aShowExtInfo;
	TPopupMenu *PopupMenu;
	TMenuItem *aDeleteAll;
	TsCheckBox *FastStatsCheckBox;
	TIdThreadComponent *RebuildXMLThreadComponent;
	TsProgressBar *RebuildXMLProgressBar;
	TsLabel *RebuildXMLLabel;
	TImage *ProgressImage;
	TsPanel *ProgressPanel;
	TsCheckBox *FrmSendNotificationCheckBox;
	TsCheckBox *SaveInArchiveCheckBox;
	TAction *aDeleteItem;
	void __fastcall FormCreate(TObject *Sender);
	void __fastcall FormShow(TObject *Sender);
	void __fastcall aExitExecute(TObject *Sender);
	void __fastcall aLoadSettingsExecute(TObject *Sender);
	void __fastcall aSaveSettingsExecute(TObject *Sender);
	void __fastcall SaveButtonClick(TObject *Sender);
	void __fastcall OKButtonClick(TObject *Sender);
	void __fastcall aAllowSaveExecute(TObject *Sender);
	void __fastcall aGetDataFromXMLExecute(TObject *Sender);
	void __fastcall sListViewCompare(TObject *Sender, TListItem *Item1, TListItem *Item2, int Data, int &Compare);
	void __fastcall sListViewColumnClick(TObject *Sender, TListColumn *Column);
	void __fastcall StatsTabSheetShow(TObject *Sender);
	void __fastcall aShowExtInfoExecute(TObject *Sender);
	void __fastcall ExInfoCheckBoxClick(TObject *Sender);
	void __fastcall aDeleteAllClick(TObject *Sender);
	void __fastcall RebuildXMLThreadComponentRun(TIdThreadComponent *Sender);
	void __fastcall sSkinManagerSysDlgInit(TacSysDlgData DlgData, bool &AllowSkinning);
	void __fastcall sPageControlPageChanging(TObject *Sender, TsTabSheet *NewPage, bool &AllowChange);
	void __fastcall aDeleteItemExecute(TObject *Sender);
private:	// User declarations
public:		// User declarations
	__fastcall TSettingsForm(TComponent* Owner);
	bool pFastStats;
	void __fastcall WMTransparency(TMessage &Message);
	BEGIN_MESSAGE_MAP
	MESSAGE_HANDLER(WM_ALPHAWINDOWS,TMessage,WMTransparency);
	END_MESSAGE_MAP(TForm)
};
//---------------------------------------------------------------------------
extern PACKAGE TSettingsForm *SettingsForm;
//---------------------------------------------------------------------------
#endif


SettingsFrm.cpp
//---------------------------------------------------------------------------
// Copyright (C) 2013-2015 Krzysztof Grochocki
//
// This file is part of NotifyMe
//
// NotifyMe is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3, or (at your option)
// any later version.
//
// NotifyMe is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with GNU Radio; see the file COPYING. If not, write to
// the Free Software Foundation, Inc., 51 Franklin Street,
// Boston, MA 02110-1301, USA.
//---------------------------------------------------------------------------

//---------------------------------------------------------------------------
#include <vcl.h>
#include <XMLDoc.hpp>
#include <LangAPI.hpp>
#pragma hdrstop
#include "SettingsFrm.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "acPNG"
#pragma link "acProgressBar"
#pragma link "sBevel"
#pragma link "sButton"
#pragma link "sCheckBox"
#pragma link "sEdit"
#pragma link "sLabel"
#pragma link "sListView"
#pragma link "sPageControl"
#pragma link "sPanel"
#pragma link "sSkinManager"
#pragma link "sSkinProvider"
#pragma link "sSpinEdit"
#pragma resource "*.dfm"
TSettingsForm *SettingsForm;
//---------------------------------------------------------------------------
__declspec(dllimport)UnicodeString GetPluginUserDir();
__declspec(dllimport)UnicodeString GetThemeSkinDir();
__declspec(dllimport)UnicodeString GetThemeDir();
__declspec(dllimport)UnicodeString GetDefaultThemeDir();
__declspec(dllimport)bool ChkSkinEnabled();
__declspec(dllimport)bool ChkThemeAnimateWindows();
__declspec(dllimport)bool ChkThemeGlowing();
__declspec(dllimport)int GetHUE();
__declspec(dllimport)int GetSaturation();
__declspec(dllimport)int GetBrightness();
__declspec(dllimport)void LoadSettings();
__declspec(dllimport)void SaveInfoToStatsFileEx(UnicodeString JID, UnicodeString Nick, UnicodeString Type, UnicodeString Target, UnicodeString Time);
//---------------------------------------------------------------------------
__fastcall TSettingsForm::TSettingsForm(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::WMTransparency(TMessage &Message)
{
	Application->ProcessMessages();
	if(sSkinManager->Active) sSkinProvider->BorderForm->UpdateExBordersPos(true,(int)Message.LParam);
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::FormCreate(TObject *Sender)
{
	//Lokalizowanie formy
	LangForm(this);
	//Wlaczona zaawansowana stylizacja okien
	if(ChkSkinEnabled())
	{
		UnicodeString ThemeSkinDir = GetThemeSkinDir();
		//Plik zaawansowanej stylizacji okien istnieje
		if(FileExists(ThemeSkinDir + "\\\\Skin.asz"))
		{
			//Dane pliku zaawansowanej stylizacji okien
			ThemeSkinDir = StringReplace(ThemeSkinDir, "\\\\", "\\", TReplaceFlags() << rfReplaceAll);
			sSkinManager->SkinDirectory = ThemeSkinDir;
			sSkinManager->SkinName = "Skin.asz";
			//Ustawianie animacji AlphaControls
			if(ChkThemeAnimateWindows()) sSkinManager->AnimEffects->FormShow->Time = 200;
			else sSkinManager->AnimEffects->FormShow->Time = 0;
			sSkinManager->Effects->AllowGlowing = ChkThemeGlowing();
			//Zmiana kolorystyki AlphaControls
			sSkinManager->HueOffset = GetHUE();
			sSkinManager->Saturation = GetSaturation();
			sSkinManager->Brightness = GetBrightness();
			//Aktywacja skorkowania AlphaControls
			sSkinManager->Active = true;
		}
		//Brak pliku zaawansowanej stylizacji okien
		else sSkinManager->Active = false;
	}
	//Zaawansowana stylizacja okien wylaczona
	else sSkinManager->Active = false;
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::FormShow(TObject *Sender)
{
	//Odczyt ustawien
	aLoadSettings->Execute();
	//Wylaczenie przycisku "Zastosuj"
	SaveButton->Enabled = false;
	//Pokazanie karty statystyk
	if(pFastStats) sPageControl->ActivePage = StatsTabSheet;
	//Pokazanie karty ustawien
	else sPageControl->ActivePage = SettingsTabSheet;
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::aExitExecute(TObject *Sender)
{
	Close();
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::aLoadSettingsExecute(TObject *Sender)
{
	TIniFile *Ini = new TIniFile(GetPluginUserDir() + "\\\\NotifyMe\\\\Settings.ini");
	OnVersionCheckBox->Checked = Ini->ReadBool("Settings","OnVersion",true);
	OnLastCheckBox->Checked = Ini->ReadBool("Settings","OnLast",true);
	CloudTimeOutSpinEdit->Value = Ini->ReadInteger("Settings","CloudTimeOut",6);
	FrmSendNotificationCheckBox->Checked = Ini->ReadBool("Settings","FrmSendNotification",true);
	SaveInArchiveCheckBox->Checked = Ini->ReadBool("Settings","SaveInArchive",false);
	StatsCheckBox->Checked = Ini->ReadBool("Settings","Stats",false);
	StatsTabSheet->Enabled = StatsCheckBox->Checked;
	FastStatsCheckBox->Enabled = StatsCheckBox->Checked;
	FastStatsCheckBox->Checked = Ini->ReadBool("Settings","FastStats",false);
	ExInfoCheckBox->Checked = Ini->ReadBool("Settings","ExInfo",false);
	delete Ini;
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::aSaveSettingsExecute(TObject *Sender)
{
	TIniFile *Ini = new TIniFile(GetPluginUserDir()+"\\\\NotifyMe\\\\Settings.ini");
	Ini->WriteBool("Settings","OnVersion",OnVersionCheckBox->Checked);
	Ini->WriteBool("Settings","OnLast",OnLastCheckBox->Checked);
	Ini->WriteInteger("Settings","CloudTimeOut",CloudTimeOutSpinEdit->Value);
	Ini->WriteBool("Settings","FrmSendNotification",FrmSendNotificationCheckBox->Checked);
	Ini->WriteBool("Settings","SaveInArchive",SaveInArchiveCheckBox->Checked);
	Ini->WriteBool("Settings","Stats",StatsCheckBox->Checked);
	Ini->WriteBool("Settings","FastStats",FastStatsCheckBox->Checked);
	Ini->WriteBool("Settings","ExInfo",ExInfoCheckBox->Checked);
	delete Ini;
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::SaveButtonClick(TObject *Sender)
{
	//Zapisanie ustawien
	aSaveSettings->Execute();
	//Odczyt ustawien w rdzeniu
	LoadSettings();
	//Wylaczanie przycisku
	SaveButton->Enabled = false;
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::OKButtonClick(TObject *Sender)
{
	//Zapisanie ustawien
	aSaveSettings->Execute();
	//Odczyt ustawien w rdzeniu
	LoadSettings();
	//Zamkniecie formy
	Close();
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::aAllowSaveExecute(TObject *Sender)
{
	SaveButton->Enabled = true;
	SaveInArchiveCheckBox->Enabled = FrmSendNotificationCheckBox->Checked;
	StatsTabSheet->Enabled = StatsCheckBox->Checked;
	FastStatsCheckBox->Enabled = StatsCheckBox->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::aGetDataFromXMLExecute(TObject *Sender)
{
	sListView->Clear();
	if(FileExists(GetPluginUserDir()+"\\\\NotifyMe\\\\Stats.xml"))
	{
		_di_IXMLDocument XMLDoc = NewXMLDocument();
		XMLDoc->LoadFromFile(GetPluginUserDir()+"\\\\NotifyMe\\\\Stats.xml");
		_di_IXMLNode MainNode = XMLDoc->DocumentElement;
		int NodesCount = MainNode->ChildNodes->GetCount();
		for(int Count=0;Count<NodesCount;Count++)
		{
			_di_IXMLNode ChildNodes = MainNode->ChildNodes->GetNode(Count);
			sListView->Items->Add();
			sListView->Items->Item[sListView->Items->Count-1]->Caption = ChildNodes->Attributes["nick"];
			sListView->Items->Item[sListView->Items->Count-1]->SubItems->Add(ChildNodes->Attributes["jid"]);
			if(ChildNodes->Attributes["type"]=="1") sListView->Items->Item[sListView->Items->Count-1]->SubItems->Add(GetLangStr("Version"));
			else sListView->Items->Item[sListView->Items->Count-1]->SubItems->Add(GetLangStr("Activity"));
			sListView->Items->Item[sListView->Items->Count-1]->SubItems->Add(ChildNodes->Attributes["time"]);
			sListView->Items->Item[sListView->Items->Count-1]->SubItems->Add(ChildNodes->Attributes["target"]);
		}
	}
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::sListViewColumnClick(TObject *Sender, TListColumn *Column)
{
	static bool SortParam = false;
	((TsListView *)Sender)->Tag = Column->Index;
	((TsListView *)Sender)->CustomSort(NULL, (long)SortParam);
	SortParam = !SortParam;
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::sListViewCompare(TObject *Sender, TListItem *Item1, TListItem *Item2, int Data, int &Compare)
{
	if(((TsListView *)Sender)->Tag==0)
	{
		Compare = CompareText(Item1->Caption, Item2->Caption);
	}
	else
	{
		int Tag = ((TsListView *)Sender)->Tag - 1;
		Compare = CompareText(Item1->SubItems->Strings[Tag], Item2->SubItems->Strings[Tag]);
	}
	if(Data==1) Compare = -Compare;
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::StatsTabSheetShow(TObject *Sender)
{
	//Pobranie danych z pliku XML
	aGetDataFromXML->Execute();
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::aShowExtInfoExecute(TObject *Sender)
{
	//Aktywna karta ustawien
	if(sPageControl->ActivePage==SettingsTabSheet)
	{
		//Szerokosc formy
		Width = 332;
		//Wysokosc formy
		Height = 327;
	}
	//Aktywna karta statystyk
	else
	{
		//Szczegolowe informacje
		if(ExInfoCheckBox->Checked)
		{
			//Margines panelu postepu
			ProgressPanel->Margins->Top = 52;
			//Szerokosc formy
			Width = 512;
			//Wysokosc formy
			Height = 412;
			//Szerokosc kolumn
			sListView->Column[1]->Width = 90;
			sListView->Column[4]->Width = 90;
		}
		//Brak szczegolowych informacji
		else
		{
		//Margines panelu postepu
		ProgressPanel->Margins->Top = 12;
		//Szerokosc formy
		Width = 332;
		//Wysokosc formy
		Height = 327;
		//Szerokosc kolumn
		sListView->Column[1]->Width = 0;
		sListView->Column[4]->Width = 0;
		}
	}
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::ExInfoCheckBoxClick(TObject *Sender)
{
	//Pokazywanie dodatkowych danych
	aShowExtInfo->Execute();
	//Zezwolenie na zapisanie ustawien
	aAllowSave->Execute();
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::aDeleteAllClick(TObject *Sender)
{
	if(FileExists(GetPluginUserDir()+"\\\\NotifyMe\\\\Stats.xml"))
	{
		DeleteFile(GetPluginUserDir()+"\\\\NotifyMe\\\\Stats.xml");
		sListView->Clear();
	}
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::RebuildXMLThreadComponentRun(TIdThreadComponent *Sender)
{
	//Usuwanie pliku XML
	if(FileExists(GetPluginUserDir()+"\\\\NotifyMe\\\\Stats.xml"))
		DeleteFile(GetPluginUserDir()+"\\\\NotifyMe\\\\Stats.xml");
	//Zapisywanie ponownie wszystkich danych do pliku XML
	for(int Count=0;Count<sListView->Items->Count;Count++)
	{
		//Pobieranie danych z listy i parsowanie
		UnicodeString Nick = sListView->Items->Item[Count]->Caption;
		UnicodeString JID = sListView->Items->Item[Count]->SubItems->Strings[0];
		UnicodeString Type = sListView->Items->Item[Count]->SubItems->Strings[1];
		if(Type==GetLangStr("Version")) Type = "1";
		else Type = "2";
		UnicodeString Time = sListView->Items->Item[Count]->SubItems->Strings[2];
		UnicodeString Target = sListView->Items->Item[Count]->SubItems->Strings[3];
		//Zapisywanie danych
		SaveInfoToStatsFileEx(JID,Nick,Type,Target,Time);
		//Dodanie pozycji do paska postepu
		RebuildXMLProgressBar->Position++;
	}
	//Pokazanie listy
	sListView->Visible = true;
	//Wlaczenie komponentow
	SettingsTabSheet->Enabled = true;
	ExInfoCheckBox->Enabled = true;
	DeleteButton->Enabled = true;
	ReloadButton->Enabled = true;
	//Zatrzymanie watku
	RebuildXMLThreadComponent->Terminate();
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::sSkinManagerSysDlgInit(TacSysDlgData DlgData, bool &AllowSkinning)
{
	AllowSkinning = false;
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::sPageControlPageChanging(TObject *Sender, TsTabSheet *NewPage, bool &AllowChange)
{
	//Pokazywanie dodatkowych danych
	aShowExtInfo->Execute();
}
//---------------------------------------------------------------------------

void __fastcall TSettingsForm::aDeleteItemExecute(TObject *Sender)
{
	//Zaznaczono na liscie jakis element
	if(sListView->ItemIndex!=-1)
	{
		//Wylaczenie komponentow
		SettingsTabSheet->Enabled = false;
		ExInfoCheckBox->Enabled = false;
		DeleteButton->Enabled = false;
		ReloadButton->Enabled = false;
		//Usuwanie wybranego elementu z listy
		sListView->Items->Item[sListView->ItemIndex]->Delete();
		//Wczytanie grafiki postepu z aktywnej kompozycji
		if(FileExists(GetThemeDir()+"////Graphics////ShortInfoURL.png"))
		 ProgressImage->Picture->LoadFromFile(GetThemeDir()+"////Graphics////ShortInfoURL.png");
		else if(FileExists(GetDefaultThemeDir()+"////Graphics////ShortInfoURL.png"))
		 ProgressImage->Picture->LoadFromFile(GetDefaultThemeDir()+"////Graphics////ShortInfoURL.png");
		//Ustawienie paska postepu
		RebuildXMLProgressBar->Position = 0;
		RebuildXMLProgressBar->Max = sListView->Items->Count;
		if(ExInfoCheckBox->Checked)	ProgressPanel->Margins->Top = 52;
		else ProgressPanel->Margins->Top = 12;
		//Ukrycie listy
		sListView->Visible = false;
		//Wlaczenie watku przebudowy pliku XML
		RebuildXMLThreadComponent->Start();
	}
}
//---------------------------------------------------------------------------

